src/blocks/utils.ts

```diff
@@ -1,8 +1,44 @@
 import { SrcId } from './types';
+import {
+  BlockFrontMatterProps,
+  ProjectFrontMatterProps,
+  DEFAULT_BLOCK_FRONT_MATTER,
+  DEFAULT_PROJECT_FRONT_MATTER,
+} from './types/frontMatter';
 
 export const srcIdToJson = (object: SrcId): SrcId => ({
   project: object.project,
   block: object.block,
   version: object.version ?? null,
   draft: object.draft ?? null,
 });
+
+const BLOCK_FRONT_MATTER_KEYS = Object.keys(
+  DEFAULT_BLOCK_FRONT_MATTER,
+) as (keyof BlockFrontMatterProps)[];
+
+const PROJECT_FRONT_MATTER_KEYS = Object.keys(
+  DEFAULT_PROJECT_FRONT_MATTER,
+) as (keyof ProjectFrontMatterProps)[];
+
+function extractFrontMatter<T extends BlockFrontMatterProps | ProjectFrontMatterProps>(
```

Comment:
add a comment to the purpose of this function?
If this just to validate keys on the json response and copy known keys from the objects only?

If so then would a list of keys be better than the default objects e.g. BLOCK_FRONT_MATTER_KEYS which may be misused? or maybe get rid of the word DEFAULT if these objects are solely there to be used to gather keys for these functions?

Reply:
e.g. something like

```
const BlockFrontMatterKeys: keyof BlockFrontMatterProps = Object.keys({
  authors: [],
  licenses: { content: '' },
  doi: '',
  open_access: false,
  github: '',
  binder: '',
  subtitle: '',
  short_title: '',
})
```

Also just realized that this function won't parse any of the nested keys e.g. `licenses: { content: 'MIT' }`